File: source/app/randomreads/lib/models/readitem.dart
class ReadItem {
  final String id;
  final String title;
  final String slug;
  final String content;
  String author = "";
  bool isAiGenerated = true;
  final String topic;
  final DateTime createdAt;
  final DateTime updatedAt;

  ReadItem({
    required this.id,
    required this.title,
    required this.slug,
    required this.content,
    required this.topic,
    required this.createdAt,
    String? author,
    bool? isAiGenerated,
  })  : author = author ?? "",
        isAiGenerated = isAiGenerated ?? true,
        updatedAt = DateTime.now();

  factory ReadItem.fromJson(Map<String, dynamic> json) {
    return ReadItem(
      id: json['id'],
      title: json['title'],
      slug: json['slug'],
      content: json['content'],
      topic: json['topic'],
      createdAt: DateTime.parse(json['createdAt']),
      author: json['author'],
      isAiGenerated: json['isAiGenerated'],
    );
  }
}

File: source/backend/RandomReads/models/ReadItem.cs

using RandomReads.Models;

public record ReadItem : CosmosItem
{
    public readonly string Title;
    public readonly string Content;
    public double rand = new Random().Next(0,1);
    public string Author { get; set; } = "";
    public string Slug => ContentUtils.GenerateSlug(Title);
    public bool IsAiGenerated { get; set; } = true;
    public readonly Topic Topic;
    public readonly DateTime CreatedAt;
    public readonly DateTime UpdatedAt = DateTime.UtcNow;

    public ReadItem(
        string Id,
        string title,
        string content,
        Topic topic,
        DateTime createdAt)
    {
        this.Id = Id;
        Title = title;
        Content = content;
        Topic = topic;
        CreatedAt = createdAt;
    }
}
